scripts/install_submitter.py

```
@@ -0,0 +1,768 @@
+
+    def set_deadline_vred_modules_env_var(self, install_dir: Path) -> bool:
+        """Set the DEADLINE_VRED_MODULES environment variable."""
+        env_var_name = ENV_VAR_DEADLINE_VRED_MODULES
+        install_path = str(install_dir)
+
+        return self._set_windows_environment_variable(env_var_name, install_path)
+
+    def install_plugin_to_vred_versions(self, install_dir: Path) -> bool:
+        """Install plugin to VRED site-packages directories."""
+        plugin_source = install_dir / "plug-ins" / "DeadlineCloudForVRED.py"
+
+        if not plugin_source.exists():
+            logger.error("Plugin file not found: %s", plugin_source)
+            return False
+
+        success_count = 0
+        vred_versions_paths = self._get_vred_installation_paths()
+
+        for version_name, vred_path in vred_versions_paths.items():
+            if vred_path and vred_path.exists():
+                site_packages = vred_path / "lib" / "python" / "Lib" / "site-packages"
+                if site_packages.exists():
+                    try:
+                        plugin_dest = site_packages / "DeadlineCloudForVRED.py"
+                        shutil.copy2(plugin_source, plugin_dest)
+                        logger.info("Installed plugin to %s: %s", version_name, plugin_dest)
+                        success_count += 1
+                    except (OSError, IOError) as e:
+                        logger.warning("Failed to install plugin to %s: %s", version_name, e)
+                else:
+                    logger.warning(
+                        "Site-packages not found for %s: %s", version_name, site_packages
+                    )
+            else:
+                logger.debug("VRED %s not found at: %s", version_name, vred_path)
+
+        if success_count > 0:
+            logger.info("Plugin installed to %d VRED version(s)", success_count)
+            return True
+
+        logger.warning("Plugin not installed to any VRED versions")
```

Question: I've noticed there's some functions we have here that return False when an issue arises as opposed to raising an exception. Examples include this function and line 188 in `copy_files`. Is this intentional?

---

Thanks, good point - I've updated the error handling to be more consistent. Those functions like `copy_files()`, `set_deadline_vred_modules_env_var()`, and `install_plugin_to_vred_versions()` now raise exceptions instead of returning False, since these are essential steps for installation. Some optional operations like `configure_vred_preferences()` still return boolean since this is the last step and users can manually configure VRED if auto-configuration fails. The install() method catches all exceptions at the top level and returns False.